### PR TITLE
Proposal: @threlte/extras `<RoundedBoxGeometry>` component

### PR DESCRIPTION
**Pull request by michealparks**
### Overview

This PR is a ready-to-merge proposal for `<RoundedBoxGeometry>`, which I discovered I wanted while attempting a [Beatsaber clone](https://github.com/michealparks/threlte-xr/blob/main/src/routes/bonksaber/scene.svelte#L98). It is roughly a port of Drei's [`<RoundedBox>`](https://github.com/pmndrs/drei#roundedbox), but with the component only encapsulating the geometry instead of the mesh so that that instanced meshes may also be used 🎉 

Happy to change any APIs that can be improved and no worries if this doesn't make sense for `@threlte/extras`!

### CODE REVIEW

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<script lang='ts'>
+
+import { Shape } from 'three'
+import { T } from '@threlte/core'
+import { toCreasedNormals } from './toCreasedNormals'
+
+export let args: [width?: number, height?: number, depth?: number] | [] = []
+export let radius = 0.05
+export let smoothness = 4
+export let steps = 1
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:10]: Undocumented prop

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<script lang='ts'>
+
+import { Shape } from 'three'
+import { T } from '@threlte/core'
+import { toCreasedNormals } from './toCreasedNormals'
+
+export let args: [width?: number, height?: number, depth?: number] | [] = []
+export let radius = 0.05
+export let smoothness = 4
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:9]: Default value not documented

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export let steps = 1
+export let creaseAngle = 0.4
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:11]: Default value not documented

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  shape.absarc(eps, eps, eps, -Math.PI / 2, -Math.PI, true)
+  shape.absarc(eps, height - radius * 2, eps, Math.PI, Math.PI / 2, true)
+  shape.absarc(width - radius * 2, height - radius * 2, eps, Math.PI / 2, 0, true)
+  shape.absarc(width - radius * 2, eps, eps, 0, -Math.PI / 2, true)
+  return shape
+}
+
+let width = 0
+let height = 0
+let depth = 0
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:27]: Initial value of `0` is different to default array assignment value of `1`

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+$: params = {
+  depth: depth - radius * 2,
+  bevelEnabled: true,
+  bevelSegments: smoothness * 2,
+  steps,
+  bevelSize: radius - eps,
+  bevelThickness: radius,
+  curveSegments: smoothness,
+}
+
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:41]: Implement events forwarding:
```svelte
<script>
  import { T, forwardEventHandlers } from '@threlte/core'
  const component = forwardEventHandlers()
</script>

<T … bind:this={$component} />
```

File: packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+</script>
+
+<T.ExtrudeGeometry
+  args={[shape, params]}
+  on:create={({ ref }) => {
+    ref.center()
+    toCreasedNormals(ref, creaseAngle)
+  }}
+/>
```
Comment (grischaerbe) [inline packages/extras/src/lib/components/RoundedBoxGeometry/RoundedBoxGeometry.svelte:50]: Add `<slot {ref} />` where `ref` is pulled from `let:ref`.

**Review by grischaerbe — CHANGES_REQUESTED** [review]
Hey @michealparks and again for your commitment, greatly appreciated!

I have some smaller remarks about the component but other than that LGTM :)

Also: please format your code with prettier before commiting 👍

**Comment by michealparks** [comment]
@grischaerbe I think I've addressed all issues, lmk if there's anything else you notice that needs fixing!

**Pull request merged**